pkg/config/identity/api_key.go

```diff
@@ -2,14 +2,59 @@ package identity
 
 import (
 	"context"
-
+	"fmt"
 	"github.com/3scale-labs/authorino/pkg/common"
+	envoy_service_auth_v3 "github.com/envoyproxy/go-control-plane/envoy/service/auth/v3"
+	ctrl "sigs.k8s.io/controller-runtime"
 )
 
 type APIKey struct {
-	SecretKey string `yaml:"secret_key"`
+	Name           string            `yaml:"name"`
+	Location       string            `yaml:"location"`
+	KeySelector    string            `yaml:"key_selector"`
+	LabelSelectors map[string]string `yaml:"label_selectors"`
+	secrets        []string
+}
+
+const (
+	locationHeader     = "header"
```

Comment:
[This page](https://www.golangprograms.com/go-language/constants.html) claims uppercase to be the convention for constant names in Golang, while [this answer](https://stackoverflow.com/a/22688926/841699) in StackOverflow points to camel case just like other symbols (variables, function names, etc).

I know the choice affects the visibility of the symbol and also that so far we seem to have adopted camel case. Still wondering if you guys are happy with this convention or not. Particularly, I'd love to use uppercase but at the same time I don't it to feel weird for other more experienced Golangers reading the code later. What do you guys think?

Reply:
The official [Golang blog](https://blog.golang.org/constants) follows the [StackOverflow answer](https://stackoverflow.com/a/22688926/841699) you found.